### ISSUE TITLE
java.lang.IllegalArgumentException: No such service ComponentInfo{com.trackees.tracks/com.transistorsoft.tsbackgroundfetch.FetchJobService}

### ISSUE THREAD

**Issue by navenduk75** [transistorsoft/flutter_background_fetch#13]
I am new to flutter. Getting below error when trying to use this plugin (Not using headless). Followed all steps mentioned in doc.

```D/TSBackgroundFetch(24484): - start
E/MethodChannel#com.transistorsoft/flutter_background_fetch/methods(24484): Failed to handle method call
 **E/MethodChannel#com.transistorsoft/flutter_background_fetch/methods(24484): java.lang.IllegalArgumentException: No such service ComponentInfo{com.trackees.tracks/com.transistorsoft.tsbackgroundfetch.FetchJobService}
E/MethodChannel#com.transistorsoft/flutter_background_fetch/methods(24484): 	at android.os.Parcel.readException(Parcel.java:2008)**
E/MethodChannel#com.transistorsoft/flutter_background_fetch/methods(24484): 	at android.os.Parcel.readException(Parcel.java:1950)
E/MethodChannel#com.transistorsoft/flutter_background_fetch/methods(24484): 	at android.app.job.IJobScheduler$Stub$Proxy.schedule(IJobScheduler.java:180)
E/MethodChannel#com.transistorsoft/flutter_background_fetch/methods(24484): 	at android.app.JobSchedulerImpl.schedule(JobSchedulerImpl.java:44)
E/MethodChannel#com.transistorsoft/flutter_background_fetch/methods(24484): 	at com.transistorsoft.tsbackgroundfetch.BackgroundFetch.start(BackgroundFetch.java:95)

**Comment by christocracy** [comment]
Show me your config.

**Comment by navenduk75** [comment]
Which file? I don't see any config file

**Comment by christocracy** [comment]
The **config you're providing to the plugin**

**Comment by navenduk75** [comment]
Future<void> initPlatformState() async {
    BackgroundFetch.configure(
            BackgroundFetchConfig(
                minimumFetchInterval: int.parse(widget.positionUpdateFrequency),
                stopOnTerminate: true,
                enableHeadless: false,
                forceReload: false),
            _onBackgroundFetch)
        .then((int status) {
      _log('[BackgroundFetch] SUCCESS: $status');
    }).catchError((e) {
      _log('[BackgroundFetch] ERROR: $e');
    });

    if (!mounted) return;
  }

**Comment by christocracy** [comment]
Syntax highlight *multiple lines* requires 3-backticks

**Comment by navenduk75** [comment]
`  Future<void> initPlatformState() async {
    BackgroundFetch.configure(
            BackgroundFetchConfig(
                minimumFetchInterval: int.parse(widget.positionUpdateFrequency),
                stopOnTerminate: true,
                enableHeadless: false,
                forceReload: false),
            _onBackgroundFetch)
        .then((int status) {
      _log('[BackgroundFetch] SUCCESS: $status');
    }).catchError((e) {
      _log('[BackgroundFetch] ERROR: $e');
    });

    if (!mounted) return;
  }`

**Comment by navenduk75** [comment]
```  Future<void> initPlatformState() async {
    BackgroundFetch.configure(
            BackgroundFetchConfig(
                minimumFetchInterval: int.parse(widget.positionUpdateFrequency),
                stopOnTerminate: true,
                enableHeadless: false,
                forceReload: false),
            _onBackgroundFetch)
        .then((int status) {
      _log('[BackgroundFetch] SUCCESS: $status');
    }).catchError((e) {
      _log('[BackgroundFetch] ERROR: $e');
    });

    if (!mounted) return;
  }

**Comment by christocracy** [comment]
You can **EDIT** an existing post.  Delete your garbage, please.

**Comment by navenduk75** [comment]
Do you have any sample flutter app using this library, not the one included in the project itself (under /example folder)?

Here is what I did:
1. Installed background_fetch (got the package)
2. Updated android/app/src/main/AndroidManifest.xml file as per doc. (Not updated any java file as enableHeadless: false, as per doc)
3. Added above code (previous comment)

I start the app, go to the page where I am using this library and I see this error.

**Comment by christocracy** [comment]
I see the problem.  You're running this on device with < API 21.

The plugin shouldn't reference [HeadlessJobService.class](https://github.com/transistorsoft/flutter_background_fetch/blob/master/android/src/main/java/com/transistorsoft/flutter/backgroundfetch/BackgroundFetchPlugin.java#L115), it should set this as a `String` instead.

`JobService` class wasn't introduced until API 21.  it should be done like this:

```java
config.setJobService("com.transistorsoft.flutter.backgroundfetch.HeadlessJobService");
```

**Comment by christocracy** [comment]
Fixed in `0.1.2`
